app/src/scripts/soc/soc.tsx

```diff
@@ -217,6 +245,16 @@ export class SOC_API extends SOC_Generic {
         lcn = 0,
     ): Promise<void> {
         if (!phrase) return Promise.resolve();
+        // Check cache for individual course code to avoid refetching
+        const upperPhrase = phrase.toUpperCase();
+        if (
+            searchBy === SearchBy.COURSE_CODE &&
+            this.courseCache.has(upperPhrase)
+        ) {
+            console.log("Course already fetched and cached:", upperPhrase);
+            // Exit if the specific course is already cached
+            return Promise.resolve();
+        }
```

Comment:
`fetchCourse(...)` should only be called when needed, hence you don't need to check if the thing you're looking for is cached (i.e. it is the responsibility of the calling function to determine whether or not to call `fetchCourses` to fetch).